File: src/main/java/com/example/onemoretimejava/arrays/KidsWithTheGreatestNumberOfCandies.java
package com.example.onemoretimejava.arrays;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class KidsWithTheGreatestNumberOfCandies {
    public static void main(String[] args) {
        /*Input: candies = [2,3,5,1,3], extraCandies = 3
        Output: [true,true,true,false,true]*/
        int[] arr = {2, 3, 5, 1, 3};
        int n = 3;
        kidsWithCandies(arr, n);

    }

    public static List<Boolean> kidsWithCandies(int[] arr, int n) {
        int len = arr.length;
        Boolean[] resArr = new Boolean[len];
        for (int i = 0; i < len; i++) {
            boolean canHaveMaxCandies = true;

            for (int j = 0; j < len; j++) {
                if (arr[i] + n < arr[j]) {
                    canHaveMaxCandies = false;
                    break;
                }
            }
            resArr[i] = canHaveMaxCandies;

        }
        System.out.println(Arrays.toString(resArr));
        return List.of(resArr);

    }
}

File: src/main/java/com/example/onemoretimekotlin/arrays/KidsWithTheGreatestNumberOfCandies.kt
package com.example.onemoretimekotlin.arrays

import java.util.Arrays


fun main(args: Array<String>) {
    /*Input: candies = [2,3,5,1,3], extraCandies = 3
       Output: [true,true,true,false,true]*/
    val arr = intArrayOf(2, 3, 5, 1, 3)
    val n = 3
    kidsWithCandies(arr, n)
}

fun kidsWithCandies(arr: IntArray, n: Int): List<Boolean> {
    val len = arr.size
    val resArr = BooleanArray(len)
    for (i in 0 until len) {
        var ss = true
        for (j in 0 until len) {
            if (arr[i] + n < arr[j]) {
                ss = false;
                break
            }
        }
        resArr[i] = ss
    }
    println(resArr.contentToString())
    return resArr.toList()

}